Since gcc6 is anal about narrowing the 0x8? constants would need
to be converted to the negative (signed char) value.
Change the 0x?? constants to characters by using `\x??' notation.

--- dcraw.c	2016-09-14 20:29:18.250684909 +0200
+++ dcraw.c	2016-09-14 20:30:37.319853026 +0200
@@ -4422,28 +4422,28 @@
 void CLASS vng_interpolate()
 {
   static const signed char *cp, terms[] = {
-    -2,-2,+0,-1,0,0x01, -2,-2,+0,+0,1,0x01, -2,-1,-1,+0,0,0x01,
-    -2,-1,+0,-1,0,0x02, -2,-1,+0,+0,0,0x03, -2,-1,+0,+1,1,0x01,
-    -2,+0,+0,-1,0,0x06, -2,+0,+0,+0,1,0x02, -2,+0,+0,+1,0,0x03,
-    -2,+1,-1,+0,0,0x04, -2,+1,+0,-1,1,0x04, -2,+1,+0,+0,0,0x06,
-    -2,+1,+0,+1,0,0x02, -2,+2,+0,+0,1,0x04, -2,+2,+0,+1,0,0x04,
-    -1,-2,-1,+0,0,0x80, -1,-2,+0,-1,0,0x01, -1,-2,+1,-1,0,0x01,
-    -1,-2,+1,+0,1,0x01, -1,-1,-1,+1,0,0x88, -1,-1,+1,-2,0,0x40,
-    -1,-1,+1,-1,0,0x22, -1,-1,+1,+0,0,0x33, -1,-1,+1,+1,1,0x11,
-    -1,+0,-1,+2,0,0x08, -1,+0,+0,-1,0,0x44, -1,+0,+0,+1,0,0x11,
-    -1,+0,+1,-2,1,0x40, -1,+0,+1,-1,0,0x66, -1,+0,+1,+0,1,0x22,
-    -1,+0,+1,+1,0,0x33, -1,+0,+1,+2,1,0x10, -1,+1,+1,-1,1,0x44,
-    -1,+1,+1,+0,0,0x66, -1,+1,+1,+1,0,0x22, -1,+1,+1,+2,0,0x10,
-    -1,+2,+0,+1,0,0x04, -1,+2,+1,+0,1,0x04, -1,+2,+1,+1,0,0x04,
-    +0,-2,+0,+0,1,0x80, +0,-1,+0,+1,1,0x88, +0,-1,+1,-2,0,0x40,
-    +0,-1,+1,+0,0,0x11, +0,-1,+2,-2,0,0x40, +0,-1,+2,-1,0,0x20,
-    +0,-1,+2,+0,0,0x30, +0,-1,+2,+1,1,0x10, +0,+0,+0,+2,1,0x08,
-    +0,+0,+2,-2,1,0x40, +0,+0,+2,-1,0,0x60, +0,+0,+2,+0,1,0x20,
-    +0,+0,+2,+1,0,0x30, +0,+0,+2,+2,1,0x10, +0,+1,+1,+0,0,0x44,
-    +0,+1,+1,+2,0,0x10, +0,+1,+2,-1,1,0x40, +0,+1,+2,+0,0,0x60,
-    +0,+1,+2,+1,0,0x20, +0,+1,+2,+2,0,0x10, +1,-2,+1,+0,0,0x80,
-    +1,-1,+1,+1,0,0x88, +1,+0,+1,+2,0,0x08, +1,+0,+2,-1,0,0x40,
-    +1,+0,+2,+1,0,0x10
+    -2,-2,+0,-1,0,'\x01', -2,-2,+0,+0,1,'\x01', -2,-1,-1,+0,0,'\x01',
+    -2,-1,+0,-1,0,'\x02', -2,-1,+0,+0,0,'\x03', -2,-1,+0,+1,1,'\x01',
+    -2,+0,+0,-1,0,'\x06', -2,+0,+0,+0,1,'\x02', -2,+0,+0,+1,0,'\x03',
+    -2,+1,-1,+0,0,'\x04', -2,+1,+0,-1,1,'\x04', -2,+1,+0,+0,0,'\x06',
+    -2,+1,+0,+1,0,'\x02', -2,+2,+0,+0,1,'\x04', -2,+2,+0,+1,0,'\x04',
+    -1,-2,-1,+0,0,'\x80', -1,-2,+0,-1,0,'\x01', -1,-2,+1,-1,0,'\x01',
+    -1,-2,+1,+0,1,'\x01', -1,-1,-1,+1,0,'\x88', -1,-1,+1,-2,0,'\x40',
+    -1,-1,+1,-1,0,'\x22', -1,-1,+1,+0,0,'\x33', -1,-1,+1,+1,1,'\x11',
+    -1,+0,-1,+2,0,'\x08', -1,+0,+0,-1,0,'\x44', -1,+0,+0,+1,0,'\x11',
+    -1,+0,+1,-2,1,'\x40', -1,+0,+1,-1,0,'\x66', -1,+0,+1,+0,1,'\x22',
+    -1,+0,+1,+1,0,'\x33', -1,+0,+1,+2,1,'\x10', -1,+1,+1,-1,1,'\x44',
+    -1,+1,+1,+0,0,'\x66', -1,+1,+1,+1,0,'\x22', -1,+1,+1,+2,0,'\x10',
+    -1,+2,+0,+1,0,'\x04', -1,+2,+1,+0,1,'\x04', -1,+2,+1,+1,0,'\x04',
+    +0,-2,+0,+0,1,'\x80', +0,-1,+0,+1,1,'\x88', +0,-1,+1,-2,0,'\x40',
+    +0,-1,+1,+0,0,'\x11', +0,-1,+2,-2,0,'\x40', +0,-1,+2,-1,0,'\x20',
+    +0,-1,+2,+0,0,'\x30', +0,-1,+2,+1,1,'\x10', +0,+0,+0,+2,1,'\x08',
+    +0,+0,+2,-2,1,'\x40', +0,+0,+2,-1,0,'\x60', +0,+0,+2,+0,1,'\x20',
+    +0,+0,+2,+1,0,'\x30', +0,+0,+2,+2,1,'\x10', +0,+1,+1,+0,0,'\x44',
+    +0,+1,+1,+2,0,'\x10', +0,+1,+2,-1,1,'\x40', +0,+1,+2,+0,0,'\x60',
+    +0,+1,+2,+1,0,'\x20', +0,+1,+2,+2,0,'\x10', +1,-2,+1,+0,0,'\x80',
+    +1,-1,+1,+1,0,'\x88', +1,+0,+1,+2,0,'\x08', +1,+0,+2,-1,0,'\x40',
+    +1,+0,+2,+1,0,'\x10'
   }, chood[] = { -1,-1, -1,0, -1,+1, 0,+1, +1,+1, +1,0, +1,-1, 0,-1 };
   ushort (*brow[5])[4], *pix;
   int prow=8, pcol=2, *ip, *code[16][16], gval[8], gmin, gmax, sum[4];
